cmd/clusterawsadm/cmd/alpha/bootstrap/bootstrap.go

```diff
@@ -135,13 +147,15 @@ func createStackCmd() *cobra.Command {
 			stsSvc := sts.NewService(awssts.New(sess))
 			accountID, stsErr := stsSvc.AccountID()
 			if stsErr != nil {
-				return stsErr
+				fmt.Printf("Error: %v", stsErr)
+				os.Exit(401)
```

Comment:
We should not be manually exiting from within a `RunE` function or Args for that matter.

I believe what you're seeing is a result of this issue: https://github.com/spf13/cobra/issues/340. We should consider evaluating our usage of `RunE` instead of hacking our way around its rather opinionated conventions. Doing this will help new and existing contributors understand the flow of the code more easily since it's not obvious why we need `os.Exit` here.

Reply:
@chuckha Would you recommend using simply `Run` instead of `RunE`? However, `Run` does not return anything by default. And, yes, we don't need `os.Exit` here (my mistake, a `return nil` along with printing error could also do it).

Reply:
I would have to look into it more. I suspect replacing `RunE` with `Run` would be sufficient, but I can't say for sure without spending some time on it as cobra has never really been intuitive for me.

Reply:
Well, replacing `RunE` with `Run` and keeping everything else same would not work since `Run` does not return anything. That also means every `return err` will be replaced by `fmt.Printf("Error: %v", err); return nil`. And, if the convention of `Run` is to be followed over `RunE`, then subsequent `RunE`'s will be also be required to be replaced by `Run`.

Or

If we are to keep `RunE` as is for now, I can replace the `os.Exit()` calls from `Args` and other places with something like:
```
if stsErr != nil {
	fmt.Printf("Error: %v", stsErr)
	return nil
}
```
Do let me know which way would you prefer and I will make the appropriate changes.

Reply:
I think that's a reasonable suggestion, keeping RunE and replacing os.Exit from Args with return nil and printing the error. Feels like we should clean that up eventually though and not use RunE unless we want that behavior. @randomvariable what do you think about this?

Reply:
Is there going to be more discussion around the possible solution which was discussed in the above comments? Or should I push the changes?

Reply:
please push the changes

Reply:
Updated. Also used cobra's built-in function `cobra.NoArgs` for validation on number of arguments passed instead of a manual check.

Reply:
I think someone else changed it from Run to RunE to fix a different UI issue, so I don't think I'm any wiser.